src/RecyclableMemoryStream.cs

```diff
@@ -1310,13 +1429,21 @@ private BlockAndOffset GetBlockAndRelativeOffset(long offset)
             return new BlockAndOffset(blockIndex, offsetIndex);
         }
 
-        private void EnsureCapacity(long newCapacity)
+        private void EnsureCapacity(long newCapacity, bool throwOomExceptionOnOverCapacity = false)
         {
             if (newCapacity > this.memoryManager.MaximumStreamCapacity && this.memoryManager.MaximumStreamCapacity > 0)
             {
                 this.memoryManager.ReportStreamOverCapacity(this.id, this.tag, newCapacity, this.AllocationStack);
-                throw new InvalidOperationException("Requested capacity is too large: " + newCapacity + ". Limit is " +
-                                                    this.memoryManager.MaximumStreamCapacity);
+                string message = $"Requested capacity is too large: {newCapacity}. Limit is {this.memoryManager.MaximumStreamCapacity}";
+
+                if (throwOomExceptionOnOverCapacity)
```

Comment:
hmm...not sure how I feel about this. I see the expectations of the interface, and they conflict with the existing implementation. I might consider just making it OOM here officially. This is going to go into v2.0, which can have some breaking changes, and this would be one.